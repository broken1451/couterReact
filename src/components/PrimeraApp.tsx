import React, { Fragment } from "react";
import PropTypes from 'prop-types';

// valor por defecto si no se envia las props
// const PrimeraApp = ({saludo = 'goku'}:any) => {
// const PrimeraApp = ({saludo, subtitulo='soy un subtitulo'}:any) => {
const PrimeraApp = ({saludo, subtitulo}:any) => {
  const saludo2 = "Hola mundo desde una variable en reactjs";
  //   const saludo = ['true',14];
  const persona = {
    nombre: "Persona",
    edad: 18,
  };

 //console.log({props});
//   if (!saludo) {
//       throw new Error("El saludo es obligatorio");
//   }

  return (
    //   <div>
    //       <h1>Hola mundo desde reactjs jejejeejjejej q tal sapo mmgvo negro </h1>
    //       <h1>Hola mundo desde reactjs jejejeejjejej q tal sapo mmgvo negro </h1>
    //   </div>

    //   <> <---- equivale a un Fragment
    //       <h1>Hola mundo desde reactjs jejejeejjejej q tal sapo mmgvo negro </h1>
    //       <h1>Hola mundo desde reactjs jejejeejjejej q tal sapo mmgvo negro </h1>
    //   </>
    <Fragment>
      <h1> {saludo2} jejejeejjejej q tal</h1>
      {/* booleando no lo impime en ----> {}, objetos da error */}
      {/* <pre> {JSON.stringify(persona)}</pre> */}
      <p>primera pagina en reactjs </p>
      <p>props en reactjs {saludo} </p>
      <p>hola soy {saludo} </p>
      <p id='p'>{subtitulo}</p>
    </Fragment>
  );
};

// definir el tipo de propiedades de este componente  las props
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    otra: PropTypes.number // si no se coloca nada es opcional
}

// valores por defecto
PrimeraApp.defaultProps ={
    subtitulo: 'soy un subtitulo'
}

export default PrimeraApp;
