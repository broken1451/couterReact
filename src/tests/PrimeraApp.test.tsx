import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
// import '@testing-library/jest-dom/extend-expect';
import PrimeraApp from "../components/PrimeraApp";


describe('Pruebas en el archivo Primeraapp ', () => {
    // test('debe de mostrar el mensaje "Hola soy goku" ', () => {
    //     // const saludo = 'Hola soy goku';
    //     // const wrapper = render(<PrimeraApp saludo={saludo}/>) // componente de react, producto renderizado del componente de react
    //     // wrapper.getByText(saludo)
    //     // // const {getByText} = render(<PrimeraApp saludo={saludo} />) // componente de react, producto renderizado del componente de react
    //     // // const {getByText} = render(<PrimeraApp saludo={saludo} />) // componente de react, producto renderizado del componente de react
    //     // // expect( getByText(saludo)).toBeInTheDocument(); 
    //     // expect( wrapper.getByText(saludo)).toBeInTheDocument(); 
    // });

    test('debe de mostrar <PrimerApp /> correctamente" ', () => {
        const saludo = 'Hola soy goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />)
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostrar el subtitulos enviado por props ', () => {
        const saludo = 'Hola soy goku';
        const subtitulo = 'subtitlo en reactjs Hola soy el subtitulo';
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />)
        const textParrafo = wrapper.find('#p').text()
        expect(textParrafo).toBe(subtitulo) 
    });
});
