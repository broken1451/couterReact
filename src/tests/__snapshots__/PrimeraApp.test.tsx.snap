// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo Primeraapp  debe de mostrar <PrimerApp /> correctamente"  1`] = `
<Fragment>
  <h1>
     
    Hola mundo desde una variable en reactjs
     jejejeejjejej q tal
  </h1>
  <p>
    primera pagina en reactjs 
  </p>
  <p>
    props en reactjs 
    Hola soy goku
     
  </p>
  <p>
    hola soy 
    Hola soy goku
     
  </p>
  <p
    id="p"
  >
    soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en el archivo demo.tets.ts debe de mostrar <PrimerApp /> correctamente"  1`] = `
<Fragment>
  <h1>
     
    Hola mundo desde una variable en reactjs
     jejejeejjejej q tal
  </h1>
  <p>
    primera pagina en reactjs 
  </p>
  <p>
    props en reactjs 
    Hola soy goku
     
  </p>
  <p>
    hola soy 
    Hola soy goku
     
  </p>
  <p
    id="p"
  >
    soy un subtitulo
  </p>
</Fragment>
`;
